setup.cfg

```diff
@@ -25,37 +25,40 @@ keywords = pytorch, deep learning, machine learning, remote sensing, satellite i
 
 [options]
 install_requires =
-    einops<0.5
-    # fiona 1.5+ required for fiona.transform module
-    fiona~=1.5
+    # einops 0.3+ required for einops.repeat
+    einops>=0.3,<0.5
+    # fiona 1.8+ required for reading empty files
+    fiona>=1.8,<2
     # kornia 0.6.4+ required for kornia.contrib.compute_padding
     kornia>=0.6.4,<0.7
-    matplotlib<4
-    numpy<2
+    # matplotlib 3.3+ required for (H, W, 1) image support in plt.imshow
+    matplotlib>=3.3,<4
+    # numpy 1.17.2+ required by pytorch-lightning
+    numpy>=1.17.2,<2
     # omegaconf 2.1+ required for to_object method
     omegaconf~=2.1
-    # pillow 2.9+ required for height attribute
-    pillow>=2.9,<10
+    # pillow 6.2+ required by matplotlib
+    pillow>=6.2,<10
     # Required to address protocolbuffers/protobuf#10051
     protobuf<4
     # pyproj 2.2+ required for CRS object
     pyproj>=2.2,<4
-    # pytorch-lightning 1.3+ required for gradient_clip_algorithm argument to Trainer
-    pytorch-lightning~=1.3
-    # rasterio 1.0.16+ required for CRS support
-    rasterio>=1.0.16,<2
+    # pytorch-lightning 1.5.1+ required for apply_to_collection bugfix
+    pytorch-lightning>=1.5.1,<2
+    # rasterio 1.0.20+ required for out_dtype parameter of DatasetReaderBase.read
```

Comment:
Unclear if this is really needed, could just cast to different dtype after reading